### PR TITLE
Persist settings on chart toggle

### PR DESCRIPTION
**Pull request by pburnsdata**
# Issues 
#232

### CODE REVIEW

File: inst/eDISH_app/modules/renderSettings/renderSettings.R [hunk 1]
```diff
@@ -108,7 +134,7 @@ renderSettings <- function(input, output, session, data, settings, status){
   #
   # update field-level inputs if a column level setting changes
   # dependent on change in data, chart selection, or column-level input
-  ######################################################################
+  ######################3################################################
```
Comment (pburnsdata) [inline inst/eDISH_app/modules/renderSettings/renderSettings.R]: hmmmm that was not intentional

File: inst/eDISH_app/modules/renderSettings/renderSettings.R [hunk 2]
```diff
@@ -60,45 +60,71 @@ renderSettings <- function(input, output, session, data, settings, status){
...
         settings = settings(),
         setting_cat_val = "appearance",
-        charts=input$charts,
+        charts=charts,
         ns=ns
       )
     )
   })
 
+  ######### Hide Settings that are not relevant to selected charts ########
```
Comment (pburnsdata) [inline inst/eDISH_app/modules/renderSettings/renderSettings.R]: Does this need to be moved to module?

File: inst/eDISH_app/modules/renderSettings/renderSettings.R [hunk 3]
```diff
@@ -60,45 +60,72 @@ renderSettings <- function(input, output, session, data, settings, status){
   ######################################################################
 
   output$data_mapping_ui <- renderUI({
-    req(input$charts)
+    charts <- isolate(input$charts)
+    req(charts)
```
Comment (bzkrouse) [inline inst/eDISH_app/modules/renderSettings/renderSettings.R]: I think req() does require a reactive inside.  So we probably want to do ```req(isolate(input$charts)) ```

File: inst/eDISH_app/modules/renderSettings/renderSettings.R [hunk 4]
```diff
@@ -60,45 +60,72 @@ renderSettings <- function(input, output, session, data, settings, status){
...
         setting_cat_val = "appearance",
-        charts=input$charts,
+        charts=charts,
         ns=ns
       )
     )
   })
 
+  ######### Hide Settings that are not relevant to selected charts ########
+  observeEvent(input$charts,{
```
Comment (bzkrouse) [inline inst/eDISH_app/modules/renderSettings/renderSettings.R]: Good work!!  App crashes if no charts are selected, so we will need to accommodate that case here.

File: inst/eDISH_app/modules/renderSettings/renderSettings.R (hunk 3 above)
Comment (bzkrouse) [reply inst/eDISH_app/modules/renderSettings/renderSettings.R]: Just tested this out and it's fine. Ignore my comment :)

File: inst/eDISH_app/modules/renderSettings/renderSettings.R (hunk 4 above)
Comment (pburnsdata) [reply inst/eDISH_app/modules/renderSettings/renderSettings.R]: Yeah, I noticed that - it was present before the change but definitely something that needs to be resolved

**Pull request merged**